### PR TITLE
Update Axom and MFEM

### LINKED ISSUE #800: Clean Spack builds after installing
Lots of object files left behind by builds (~150 megs).  Can this be solved by an additional Spack command/functionality?

### PR DESCRIPTION
**Pull request by chapman39**
* Updates spack, axom, and mfem
* Clean spack repo after successful TPL build. Fixes #800 
* I'll be skipping updating docker containers in here since that will be done in this PR: https://github.com/LLNL/serac/pull/1074

### CODE REVIEW

**Comment by chapman39** [comment]
@jamiebramwell In this new MFEM version, the strumpack ctor being used is deprecated. Looks like there is another one that takes in a matrix object that could be used instead.

```cpp
   // Constructor with STRUMPACK matrix object and command line arguments.
   STRUMPACKSolver(STRUMPACKRowLocMatrix &A, int argc, char *argv[]);
```

```
In file included from /usr/WS2/meemee/serac/repo/src/serac/numerics/equation_solver.cpp:7:
/usr/WS2/meemee/serac/repo/src/serac/infrastructure/../../serac/numerics/equation_solver.hpp:208:53: error: 'STRUMPACKSolver' is deprecated [-Werror,-Wdeprecated-declarations]
  StrumpackSolver(int print_level, MPI_Comm comm) : strumpack_solver_(0, nullptr, comm)
                                                    ^
/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2024_02_15_11_25_36/clang-14.0.6/mfem-4.6.2-rc0-nsazjwq3gfybuqduzgsph4ygf5gq5re5/include/mfem/linalg/strumpack.hpp:234:4: note: 'STRUMPACKSolver' has been explicitly marked deprecated here
   MFEM_DEPRECATED STRUMPACKSolver(int argc, char *argv[], MPI_Comm comm)
   ^
/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2024_02_15_11_25_36/clang-14.0.6/mfem-4.6.2-rc0-nsazjwq3gfybuqduzgsph4ygf5gq5re5/include/mfem/mesh/../fem/../general/globals.hpp:120:40: note: expanded from macro 'MFEM_DEPRECATED'
#define MFEM_DEPRECATED __attribute__((deprecated))
                                       ^
In file included from /usr/WS2/meemee/serac/repo/src/serac/numerics/odes.cpp:7:
In file included from /usr/WS2/meemee/serac/repo/src/serac/infrastructure/../../serac/numerics/odes.hpp:20:
/usr/WS2/meemee/serac/repo/src/serac/infrastructure/../../serac/numerics/equation_solver.hpp:208:53: error: 'STRUMPACKSolver' is deprecated [-Werror,-Wdeprecated-declarations]
  StrumpackSolver(int print_level, MPI_Comm comm) : strumpack_solver_(0, nullptr, comm)
                                                    ^
/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2024_02_15_11_25_36/clang-14.0.6/mfem-4.6.2-rc0-nsazjwq3gfybuqduzgsph4ygf5gq5re5/include/mfem/linalg/strumpack.hpp:234:4: note: 'STRUMPACKSolver' has been explicitly marked deprecated here
   MFEM_DEPRECATED STRUMPACKSolver(int argc, char *argv[], MPI_Comm comm)
   ^
/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2024_02_15_11_25_36/clang-14.0.6/mfem-4.6.2-rc0-nsazjwq3gfybuqduzgsph4ygf5gq5re5/include/mfem/mesh/../fem/../general/globals.hpp:120:40: note: expanded from macro 'MFEM_DEPRECATED'
#define MFEM_DEPRECATED __attribute__((deprecated))
```

edit: This has been resolved

**Comment by chapman39** [comment]
Updating spack and fixing this mvapich2 error I've been getting in LiDO just to get it out of the way. Once tpls have been rebuilt this pr should be ready.

**Review by jamiebramwell — APPROVED** [review]
The strumpack change looks good to me.

File: scripts/spack/packages/mfem/package.py [hunk 1]
```diff
@@ -8,12 +8,8 @@
 
 class Mfem(BuiltinMfem):
 
-    # Note: We have a `serac-dev` branch on mfem's github that we track pending changes.
-    # Note: Make sure this sha coincides with the git submodule
-    # Note: We add a number to the end of the real version number to indicate that we have
-    #       moved forward past the release. Increment the last number when updating the commit sha.
-    version('4.6.1.1', commit='89caa3ac6dd1624540e99b47406b74454efdb231')
-    
+    version("4.6.2-rc0", commit="79b609872ab4cef720adac2f820792fe09c87065")
```
Comment (white238) [inline scripts/spack/packages/mfem/package.py:11]: 👍 wooo!

File: scripts/spack/configs/toss_4_x86_64_ib/spack.yaml [hunk 2]
```diff
@@ -78,10 +78,10 @@ spack:
     mvapich2:
       buildable: false
       externals:
-      - spec: mvapich2@2.3.6%clang@14.0.6 process_managers=slurm arch=linux-rhel8-ivybridge
-        prefix: /usr/tce/packages/mvapich2/mvapich2-2.3.6-clang-14.0.6
-      - spec: mvapich2@2.3.6%gcc@10.3.1 process_managers=slurm arch=linux-rhel8-ivybridge
-        prefix: /usr/tce/packages/mvapich2/mvapich2-2.3.6-gcc-10.3.1
+      - spec: mvapich2@2.3.7%clang@14.0.6~alloca~cuda~debug~hwloc_graphics~hwlocv2+regcache+wrapperrpath build_system=autotools ch3_rank_bits=32 fabrics=mrail file_systems=auto process_managers=slurm threads=multiple
```
Comment (white238) [inline scripts/spack/configs/toss_4_x86_64_ib/spack.yaml:81]: Do I want to know what horrors caused this expansion?

**Review by white238 — APPROVED** [review]
Thanks @chapman39 !

File: scripts/spack/configs/toss_4_x86_64_ib/spack.yaml (hunk 2 above)
Comment (chapman39) [reply scripts/spack/configs/toss_4_x86_64_ib/spack.yaml:81]: Short answer, no lol. but here is a related lido issue https://lc.llnl.gov/gitlab/topopt/lido-2.0/-/issues/401

**Pull request merged**